pygmt/clib/session.py

```diff
@@ -1359,6 +1360,62 @@ def virtualfile_from_grid(self, grid):
         with self.open_virtual_file(*args) as vfile:
             yield vfile
 
+    def virtualfile_from_data(self, data, x=None, y=None, z=None, check_kind=None):
+        """
+        Store any data inside a virtual file.
+
+        This convenience function automatically detects the kind of data passed
+        into it, and produces a virtualfile that can be passed into GMT later
+        on.
+
+        Parameters
+        ----------
+        data : str, xarray.DataArray, 2d array, or None
+            The vectors that will be included in the array. All must be of the
+            same size.
```

Comment:
The description of the `data` parameter is very confusing.

Reply:
I've changed the docstring, see if it looks ok.